src/DynamoCoreWpf/Views/Core/DynamoView.xaml.cs

```diff
@@ -322,6 +340,86 @@ private void CloseExtensionTab(TabItem tabitem)
             }
         }
 
+        private void CloseExtensionWindow(string name)
+        {
+            if (ExtensionWindows.ContainsKey(name))
+            {
+                var extension = ExtensionWindows[name];
+                extension.Close();
+                ExtensionWindows.Remove(name);
+            }
+        }
+
+        internal void UndockExtensionTab(object sender, RoutedEventArgs e)
+        {
+            var tabName = (sender as Button).DataContext.ToString();
+            UndockExtension(tabName);
+        }
+
+        /// <summary>
+        /// Undocks the extension with the given name.
+        /// Made internal for testing purposes only.
+        /// </summary>
+        /// <param name="name">Name of the extension</param>
+        internal void UndockExtension(string name)
+        {
+            var tabItem = ExtensionTabItems.OfType<TabItem>().SingleOrDefault(tab => tab.Header.ToString() == name);
+            CloseExtensionTab(tabItem);
+
+            var ext = new ExtensionWindow();
```

Comment:
Curious if we can reuse the extension window so that we do not need to always create new ones, but given it is only a frame, seems no harm yet?

Reply:
They should be GC'ed just like tabs. Recycling them would be cumbersome and wouldn't bring any benefit IMO.

Reply:
@mmisol I'm cool with this implementation for now.